src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BodyModelBinderProvider.cs

```diff
@@ -52,6 +52,17 @@ public BodyModelBinderProvider(IList<IInputFormatter> formatters, IHttpRequestSt
             _loggerFactory = loggerFactory;
         }
 
+        /// <summary>
+        /// Gets or sets a flag that controls whether the binder should require non-null
+        /// input values to be supplied. Defaults to <c>true</c>.
+        /// </summary>
+        /// <example>
+        /// If <c>true</c>, and an associated <see cref="IInputFormatter"/> supplies <c>null</c>
+        /// as the value to bind (for example, if an incoming POST or PUT request supplies a
+        /// zero-length body), the binder will register a validation error.
+        /// </example>
+        public bool IsBindingRequired { get; set; } = true;
```

Comment:
This cannot easily be set `false`.

Instead, use existing infrastructure that plumbs `ModelMetadata.IsBindRequired` through and base decisions in the `InputFormatter` on that property. Add a new `MvcOptions` property e.g. `IgnoreEmptyBodyInInputFormatter` that defaults to `false` (if you use something similar to that name). Also add an `IBindingMetadataProvider` implementation (or extend `DefaultBindingMetadataProvider`) similar to `DataMemberRequiredBindingMetadataProvider` but based on the calculated `BindingSource` and the new option.